programs/dynamic-bonding-curve/src/state/config.rs

```diff
@@ -519,6 +524,43 @@ impl PoolConfig {
                 .map_err(|_| PoolError::InvalidTokenUpdateAuthorityOption)?;
         Ok(token_update_authority)
     }
+
+    pub fn get_migration_quote_amount_for_config(&self) -> Result<MigrationAmount> {
+        PoolConfig::get_migration_quote_amount(
+            self.migration_quote_threshold,
+            self.migration_fee_percentage,
+        )
+    }
+    pub fn get_migration_quote_amount(
+        migration_quote_threshold: u64,
+        migration_fee_percentage: u8,
+    ) -> Result<MigrationAmount> {
+        let quote_amount: u64 = safe_mul_div_cast_u64(
+            migration_quote_threshold,
+            100.safe_sub(migration_fee_percentage.into())?,
+            100,
+            Rounding::Up,
+        )?;
+        let fee = migration_quote_threshold.safe_sub(quote_amount)?;
+        Ok(MigrationAmount { quote_amount, fee })
+    }
+
+    pub fn get_migration_fee_distribution(&self) -> Result<MigrationFeeDistribution> {
+        let MigrationAmount {
+            quote_amount: _quote_amount,
+            fee,
+        } = self.get_migration_quote_amount_for_config()?;
+
```

Comment:
Can early return if `self.creator_migration_fee_percentage` is 0 or 100

Reply:
that is fine to keep as the current logic